_posts/2021-04-07-#21061.md

```diff
@@ -7,19 +7,80 @@ authors: [amitiuttarwar]
 components: ["p2p"]
 host: glozow
 status: upcoming
-commit:
+commit: 038f751
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+
+* Hiding links between wallet addresses and IP addresses is a key part of
+  Bitcoin privacy. Many techniques exist to help users obfuscate their IP
+address when submitting their own transactions, and various P2P changes have
+been proposed with the goal of hiding transaction origins.
+
+* However, _rebroadcast_ behavior can also leak information. If a node
+  rebroadcasts its own wallet transactions differently from transactions
+received from its peers, adversaries can use this information to infer
+transaction origins even if the initial broadcast revealed nothing. We have
+discussed rebroadcast in previous review clubs,
+[#16698](https://bitcoincore.reviews/16698) and
+[#18038](https://bitcoincore.reviews/18038).
+
+* The rebroadcast project's goal is to improve privacy by making node
+  rebroadcast behavior for wallet transactions indistinguishable from that of
+other peers' transactions.
+
+* PR#21061 adds a `TxRebroadcast` module responsible for selecting transactions
+  to be rebroadcast and keeping track of how many times each transaction has
+been rebroadcast.  After each block, the module uses the miner and other
+heuristics to select transactions from the mempool that it believes "should"
+have been included in the block and reannounces them (disabled by default for
+now).
+
+* Rebroadcasts happen once per new bock. The set of transactions to be
+  rebroadcast is calculated as follows:
+
+	- The node regularly estimates the minimum feerate for transactions to
+	  be included in the next block, `m_cached_fee_rate`.
+
+	- When a new block arrives, the transactions included in the block are
+	  removed from the mempool. The node then uses `BlockAssembler` to
+calculate which transactions (with a total weight up to 3/4 of the block
+maximum) from the mempool are more than 30 minutes old and have a minimum
+feerate of `m_cached_fee_rate`. The combination of these two filters results in
```

Comment:
mmm, I'd say fee rate filter = "set of transactions that our node would have included in the last block", time filter is just to add some buffer / reduce mishits